plugins/modules/foreman_ansible_roles_facts.py

```diff
@@ -0,0 +1,186 @@
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: foreman_ansible_roles_facts
+short_description: Gather Ansible Roles facts
+description:
+  - Gather facts about Foreman Ansible Roles
+author:
+  - "Brant Evans (@branic)
+options:
+  action:
+    description:
+      - Method to use for gathering facts
+    default: list
+    choices:
+      - list
+      - show
+      - fetch
+    required: false
+    type: str
+  proxy:
+    description:
+      - The Smart Porxy name to use for fetching Ansible Roles
+    required: false
+    type: str
+    alias: smart_proxy
+  search:
+    description:
+      - Search query to use
+      - If None, all resources are returned
+    type: str
+  full_details:
+    description:
+      - If C(True) all details about the found resources are returned
+    type: bool
+    default: false
+    aliases: [ info ]
+  organization:
+    description:
+      - Organization that the role is in
+    required: false
+    type: str
+  location:
+    description:
+      - Location that the role is in
+    required: false
+    type: str
+extends_documentation_fragment:
+  - foreman
+'''
+
+
+EXAMPLES = '''
+- name: Get Ansible Roles available to import on a proxy
+  foreman_ansible_roles:
+    username: "admin"
+    password: "changeme"
+    server_url: "https://foreman.example.com"
+    state: "fetch"
+    proxy: "foreman.example.com"
+  register: result
+- debug:
+    var: result
+
+- name: Get all imported Ansible Roles
+  foreman_ansible_roles:
+    username: "admin"
+    password: "changeme"
+    server_url: "https://foreman.example.com"
+    organization: "
+    state: "list"
+  register: result
+- debug:
+    var: result
+
+- name: Find the imported Ansible Role named example_role
+  foreman_ansible_roles:
+    username: "admin"
+    password: "changeme"
+    server_url: "https://foreman.example.com"
+    organization: "
+    state: "list"
+    search: "name=example_role"
+  register: result
+- debug:
+    var: result
+'''
+
+RETURN = '''
+ansible_roles:
+  description: Roles that are able to be imported on the proxy
+  returned: state = fetched
+  type: list
+'''
+
+from ansible.module_utils.foreman_helper import ForemanEntityAnsibleModule
+
+
+class ForemanAnsibleRolesFactsModule(ForemanEntityAnsibleModule):
+    pass
+
+
+def main():
+
+    module = ForemanAnsibleRolesFactsModule(
+        foreman_spec = dict(
+            action = dict(default='fetch', choices=['fetch', 'list', 'show']),
+            proxy = dict(type='entity', flat_name='proxy_id', aliases=['smart_proxy'], resource_type='smart_proxies'),
```

Comment:
```suggestion
            action=dict(default='fetch', choices=['fetch', 'list', 'show']),
            proxy=dict(type='entity', flat_name='proxy_id', aliases=['smart_proxy'], resource_type='smart_proxies'),
```

the linter is picky ;)

Reply:
Fixed.